lib/disk/modules/QcowDisk.rb

```diff
@@ -571,6 +620,30 @@ def cluster_offset_mask
     @cluster_offset_mask ||= (1 << csize_shift) - 1
   end
 
+  def incompatible_features
+    version < 3 ? 0 : @header['incompatible_features']
```

Comment:
It would make for cleaner code but wouldn't be technically correct. These fields are not present in qcow version 2 and are assumed to be zero (hence the explicit distinction)

https://github.com/qemu/QEMU/blob/master/docs/specs/qcow2.txt#L71